### PR TITLE
fix `test_qnode_probs_expval_broadcasting`

### PR DESCRIPTION
**Pull request by JerryChen97**
**Context:**
Somehow, this test is simply written wrong:
1. The shot vector branch `if hasattr(shots, "len"):` shall never enter because `list` has not attr `len` anymore
2. The unpacking logic for `jac` is wrong since the first dimension should be of the shot vector and the second is for meansurmeents

**Description of the Change:**

**Benefits:**

**Possible Drawbacks:**

**Related GitHub Issues:**
[sc-101765]

### CODE REVIEW

File: tests/gradients/core/test_pulse_gradient.py [hunk 1]
```diff
@@ -1610,10 +1607,10 @@ def circuit(params):
         jac = jax.jacobian(circuit)(params)
         p = params[0] * T
         exp_jac = (jnp.array([-1, 1]) * jnp.sin(2 * p) * T, -2 * jnp.sin(2 * p) * T)
-        if hasattr(shots, "len"):
```
Comment (andrijapau) [inline tests/gradients/core/test_pulse_gradient.py:1613]: 😅 what

File: tests/gradients/core/test_pulse_gradient.py [hunk 2]
```diff
@@ -1610,10 +1607,10 @@ def circuit(params):
...
-            for j_shots, e_shots in zip(jac, exp_jac):
-                for j, e in zip(j_shots, e_shots):
-                    assert qml.math.allclose(j[0], e, atol=tol, rtol=0.0)
+        if isinstance(shots, list):
```
Comment (andrijapau) [inline tests/gradients/core/test_pulse_gradient.py:1610]: Maybe this?
```suggestion
        if shots.has_partitioned_shots:
```

Comment (JerryChen97) [reply tests/gradients/core/test_pulse_gradient.py:1610]: But the input `shots` for this test is `none`, `100`, or `[100, 100]` instead of `Shots` something 🤣

Comment (andrijapau) [reply tests/gradients/core/test_pulse_gradient.py:1610]: ohhh

Comment (andrijapau) [reply tests/gradients/core/test_pulse_gradient.py:1610]: We could just wrap it in `Shots(...)` but i see what you mean
